pkg/engine/engine.go

```diff
@@ -140,6 +143,16 @@ func (e *Engine) alterFuncMap(t *template.Template) template.FuncMap {
 		return buf.String()
 	}
 
+	// Add the 'required' function here
+	funcMap["required"] = func(warn string, val string) (string, error) {
```

Comment:
I am not very familiar with the templating engine itself, but shouldn't this return an `interface` so it doesn't convert other types to strings?